---
title: "class 10"
author: "Kalodiah Toma (A07606689)"
format: pdf
---

The main repository of sturctural data is the PDB. Let's examine what it contains. 

I downloaded composition stats from:  < https://www.rcsb.org/stats/summary >

```{r}
round(183201/251600768*100, 2)
```
```{r}
stats <-read.csv("Data Export Summary.csv", row.names = 1)
head(stats)
```
```{r}
string<-c("10", "100", 1)
as.numeric(string) 
```
```{r}
x<-string
as.numeric(gsub(",", "", x) )
rm.comma<-function(x){
  as.numeric(gsub(",", "", x) )
}
pdbstats<-apply(stats, 2, rm.comma)
```
```{r}
rownames(pdbstats)<-row.names(stats)
pdbstats
```

Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.
In UniPort there are 25,1600,768 protein sequences. 
In PDB there are 183,201 protein structures. 
0.07% of protein sequences have protein sturctures determined. 
84.83% for X-ray and 8.33% for EM of structures are in the PDB are solved by X-Ray and Electron Microscopy
```{r}
totals <- apply(pdbstats, 2, sum)
round(totals/totals["Total"] *100, 2)
```

Q2: What proportion of structures in the PDB are protein?
Skipped in class

Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?
skipped in class


1 protease is "D" amino acid in the structure of 1hsg. 
MK1 is the drug that inhibits HIV

#Using Mol* to examine HIV-Or

Here is a rubish pic of HIV-Pr that is not very useful yet. 

![](1HSG.png)
And a nicer pic colored by secondary structure with catalytic active site ASP25 shown in each chain along with MK1 drug. 

![Image caption here](1HSG1.png)
Q4: Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?
Hydrogen is 0.5A and the resolution is 2A so we do not see hydrogen in this structure. 

Q5: There is a critical “conserved” water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have
H380 is the multi billion dollar water

Q6: Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend “Ball & Stick” for these side-chains). Add this figure to your Quarto document.
See images above

#Using the bio3d package

```{r}
library(bio3d)
pdb <- read.pdb("1hsg")
pdb
```
Q7: How many amino acid residues are there in this pdb object? 
198

Q8: Name one of the two non-protein residues? 
MK1

Q9: How many protein chains are in this structure? 
2

```{r}
attributes(pdb)
head(pdb$atom)
```
```{r}
head(pdb$atom$resid)
```

```{r}
aa321( pdb$atom$resid[pdb$calpha] )
```
#Predicting functional motions of a single structure 

Run a Normal Mode Analysis (NMA) - a bioinformatics methos to predict functional motions.
```{r}
adk <-read.pdb("6s36")
adk
modes<-nma(adk)
plot(modes)
```
```{r}
mktrj(modes, pdb=adk, file="modes.pdb")
```
![](modes.pdb.png)

